server/aws-lsp-identity/src/sso/authorizationCodePkce/authorizationServer.ts

```diff
@@ -0,0 +1,156 @@
+            authReject = reject
+        })
+        this.authResolve = authResolve!
+        this.authReject = authReject!
+    }
+
+    async [Symbol.dispose]() {
+        this.httpServer.closeAllConnections()
+        this.httpServer.close()
+    }
+
+    static async start(httpServer?: Server): Promise<AuthorizationServer> {
+        httpServer ||= http.createServer()
+
+        // Wait for server to start listening
+        await new Promise<void>(resolve => {
+            httpServer.once('listening', resolve)
+            httpServer.listen(0, '127.0.0.1')
+        })
+
+        return new AuthorizationServer(httpServer)
+    }
+
+    async authorizationCode(): Promise<string> {
+        return this.authPromise
+    }
+
+    private async requesterListener(req: IncomingMessage, res: ServerResponse): Promise<void> {
+        if (!req.url) {
+            return
+        }
+
+        res.setHeader('Access-Control-Allow-Methods', 'GET')
+
+        if (new URL(req.url, this.origin).pathname === AuthorizationServer.authorizationPath) {
+            this.authorizationRequest(req, res)
+        } else {
+            this.resourceRequest(req, res)
+        }
+    }
+
+    private authorizationRequest(req: IncomingMessage, res: ServerResponse) {
+        if (!req.url) {
+            return
+        }
+
+        try {
+            const searchParams = new URL(req.url, this.origin).searchParams
+
+            const error = searchParams.get('error')
+            if (error) {
+                throw new Error(`${error}: ${searchParams.get('error_description') ?? 'No description'}`)
+            }
+
+            const code = searchParams.get('code')
+            if (!code) {
+                throw new Error('Authorization code not found.')
+            }
+
+            const state = searchParams.get('state')
+            if (!state) {
+                throw new Error('CSRF state not found.')
+            }
+
+            if (state !== this.csrfState) {
+                throw new Error('CSRF state is invalid.')
+            }
+
+            try {
+                this.redirect(res)
+            } finally {
+                this.authResolve(code)
+            }
+        } catch (error) {
+            try {
+                this.redirect(res, error?.toString())
+            } finally {
+                this.authReject(error)
+            }
+        }
+    }
+
+    private redirect(res: ServerResponse, error?: string) {
+        let redirectUri = `${this.origin}/index.html`
+        if (error) {
+            redirectUri += `?${new URLSearchParams({ error })}`
+        }
+
+        res.setHeader('Location', redirectUri)
+        res.writeHead(302)
+        res.end()
+    }
+
+    private async resourceRequest(req: IncomingMessage, res: ServerResponse): Promise<void> {
+        if (!req.url) {
+            return
+        }
+
+        try {
+            const resource = await readFile(path.join(__dirname, 'resources', new URL(req.url, this.origin).pathname))
```

Comment:
This would be relative to the authorizationServer.js file under out (due to use of __dirname).  However, see your other comment below about moving these files to a consistent location across servers.